scripts/bozzcoin.js

```diff
@@ -0,0 +1,203 @@
+// Description:
+//   Tracks bozzcoins earned through exercise
+//
+// Commands: Hubot must be specifically @-mentioned.
+//   bozzcoins? - reports how many bozzcoins are in the account.
+//   did x (pullups|pushups|situps|squats|lunges|racket steps)
+//      - adds bozzcoins to the account depending on the exercise type.
+//   ran x km - adds 100 bozzcoins per km.
+//   (prata|starbucks|macs) day - subtracts bozzcoins depending on the cheat type.
+
+module.exports = (robot) => {
+  // number of days bozz can go without exercising
+  const bozziplierThreshold = 2;
+  // multiplier decay for bozziplier
+  const bozziplierDecay = 0.8;
+
+  // produces a summary of who has contributed bozzcoins
+  function bozzcoinSummaries() {
+    const output = [];
+    const bozzcoinTracker = robot.brain.get('bozzcoinTracker');
+    const sortedBozzcoinTracker = Object.keys(bozzcoinTracker).sort((a, b) => bozzcoinTracker[b] - bozzcoinTracker[a]);
+    sortedBozzcoinTracker.forEach((name) => {
+      const bozzcoinsContributed = bozzcoinTracker[name];
+      output.push(`${(name === 'rurouni' ? `*${name}*` : name)}: ${bozzcoinsContributed} :bozzcoin: contributed.`);
+    });
+    return `\n${output.join('\n')}`;
+  }
+
+  function earnRate(exerciseType) {
+    switch (exerciseType) {
+      case 'cycle': return 20;
+      case 'pullups': return 10;
+      case 'pushups': case 'situps': case 'squats': case 'lunges': return 1;
+      case 'run': return 50;
+      case 'racket steps': return 1 / 18;
+      default: return 0;
+    }
+  }
+
+  function verbToExerciseType(verb) {
+    switch (verb) {
+      case 'ran': return 'run';
+      case 'cycled': return 'cycle';
+      default: return verb;
+    }
+  }
+
+  function bozzExercisesFrequentEnough() {
+    const thresholdInMilliSecs = bozziplierThreshold * 86400000;
+    const bozzLastExercised = robot.brain.get('bozzLastExercised');
+    return (new Date().getTime() - new Date(bozzLastExercised).getTime()) > thresholdInMilliSecs;
```

Comment:
shouldn't it be the other way round? the difference must be less than threshold?

Reply:
Oops. Good catch. Fixed.